### PR TITLE
[Step 1] Upgrade Google Auth Version

### PR DESCRIPTION
**Pull request by ThomasCJY**
upgrade google auth to use the latest v1.6.0 instead of the outdated 0.17 in bazel

pt 1: Add Gson and upgrade Auth (third_party only changes)

part of https://github.com/bazelbuild/bazel/pull/15176

### CODE REVIEW

File: third_party/BUILD [hunk 1]
```diff
@@ -163,8 +163,10 @@ distrib_java_import(
     jars = [
         "api_client/google-api-client-1.22.0.jar",
         "api_client/google-api-client-jackson2-1.22.0.jar",
```
Comment (coeuvre) [inline third_party/BUILD:165]: I think we can remove two jackson2 jars in this PR.

File: third_party/BUILD [hunk 2]
```diff
@@ -163,8 +163,10 @@ distrib_java_import(
...
+        "api_client/google-api-client-gson-1.22.0.jar",
         "api_client/google-http-client-1.22.0-SNAPSHOT.jar",
         "api_client/google-http-client-jackson2-1.22.0.jar",
+        "api_client/google-http-client-gson-1.22.0.jar",
     ],
     runtime_deps = [
```
Comment (coeuvre) [inline third_party/BUILD:171]: need `:gson` here.

**Review by coeuvre — CHANGES_REQUESTED** [review]

File: third_party/BUILD (hunk 1 above)
Comment (ThomasCJY) [reply third_party/BUILD:165]: wouldn't that require us to update test code `import com.google.api.client.json.jackson2.JacksonFactory;` as well?

File: third_party/BUILD (hunk 2 above)
Comment (ThomasCJY) [reply third_party/BUILD:171]: thanks. updated

File: third_party/BUILD (hunk 1 above)
Comment (coeuvre) [reply third_party/BUILD:165]: Right. I thought only `:jackson2` is required for the test code. Please add them back and we are ready to merge!

Comment (ThomasCJY) [reply third_party/BUILD:165]: pr updated

**Review by coeuvre — APPROVED** [review]
Thanks!

@meteorcloudy Can you help import this PR?

**Comment by meteorcloudy** [comment]
@ThomasCJY Can you squash all commits into a single one? That's easier for importing.

**Comment by ThomasCJY** [comment]
@meteorcloudy @coeuvre ^PTAL

**Pull request closed without merging**